### ISSUE TITLE
ON DELETE SET NULL (column) not supported at the moment?

### ISSUE THREAD

**Issue by joidegn** [sqlalchemy/sqlalchemy#11595]
### Describe the use case

Im trying to issue a `ON DELETE SET NULL (column)` as described in the [Postgres docs](https://www.postgresql.org/docs/16/ddl-constraints.html) since version 15. 

Im getting this error from SQLAlchemy which makes me think we don't support this yet. I also can't find this mentioned in the docs. 

I am using sqlalchemy 2.0.31.

`sqlalchemy.exc.CompileError: Unexpected SQL phrase: 'SET NULL (applied_exchange_rate_id, base_currency_amount)' (matching against '^(?:RESTRICT|CASCADE|SET NULL|NO ACTION|SET DEFAULT)$')`

### Databases / Backends / Drivers targeted

postgresql and driver psycopg2

### Example Use

```
    applied_exchange_rate_id = Column(pg_UUID, ForeignKey("exchange_rates.id", ondelete="SET NULL (applied_exchange_rate_id, base_currency_amount)"), nullable=True)
    applied_custom_exchange_rate_id = Column(pg_UUID, ForeignKey("exchange_rates.id", ondelete="SET NULL (applied_custom_exchange_rate_id, base_currency_amount)"), nullable=True)

```

**Comment by CaselIT** [comment]
Hi,

No that syntax is not supported at the moment, but PR would be accepted for it.

@zzzeek I guess we should attach the `FK_ON_DELETE` etc to the compiler so postgresql can override it?

**Comment by joidegn** [comment]
Makes sense! I have not committed code to sqlalchemy yet but I might try my hand if I find the time 🙏

**Comment by CaselIT** [comment]
It's likely that your code will probably work if you just comment this check
https://github.com/sqlalchemy/sqlalchemy/blob/772ab0befefe0dad99db22e21ed5f37afd1a1928/lib/sqlalchemy/sql/compiler.py#L7443-L7459


Reflection would probably also need to be adapted but it could be done in a second moment

**Comment by zzzeek** [comment]
oh it's in the spirit of SQL injection prevention.   which is silly as this is DDL but let's leave it

**Comment by CaselIT** [comment]
> oh it's in the spirit of SQL injection prevention. which is silly as this is DDL but let's leave it

I think so. The blame mentions CVEs.

**Comment by sanfong** [comment]
For those who stumble upon this issue like me, this is my work around.
https://github.com/sqlalchemy/sqlalchemy/blob/7949426428f1ec19381116e025b95f86417a85fc/lib/sqlalchemy/sql/compiler.py#L234-L236
I just overwrite `FK_ON_DELETE`
```py
from sqlalchemy.sql import compiler
compiler.FK_ON_DELETE = re.compile(
    r"^(?:RESTRICT|CASCADE|SET NULL|SET NULL\s?(.*)|NO ACTION|SET DEFAULT)$", re.I
)
```

**Comment by CaselIT** [comment]
see also #11946
